### DISCUSSION TITLE
Discrepancy Between Training and Validation/Test Loss in MACE Model

### DISCUSSION THREAD

**Discussion by imshivam24** [ACEsuit/mace#1036] [Q&A] [answered]
Hi everyone,

I am training the MACE model from scratch for my system, using separate files for training, validation, and test datasets. During training, the log shows a consistent decrease in validation loss for both energy and forces. However, by the end of training, the training error is several orders of magnitude higher than the validation and test errors.

I’m struggling to understand why this is happening and would appreciate any insights or suggestions. I’ve attached the log file and input commands for reference.
[MACE_model_regularized_run-1.log](https://github.com/user-attachments/files/20900761/MACE_model_regularized_run-1.log)

Thanks in advance for your help!

snippet of val.extxyz
![image](https://github.com/user-attachments/assets/fa6d3c73-8536-4d58-8907-1b534bff7d62)

command = [
    "mace_run_train",
    "--name", "MACE_model_regularized",
    "--train_file", "../../train.extxyz",
    "--valid_file", "../../val.extxyz",
    "--test_file", "../../test.extxyz",
    "--config_type_weights", '{"Default":1.0}',
    "--E0s", "average",
    "--model", "ScaleShiftMACE",
    "--energy_key", "energy",
    "--forces_key", "forces",
    "--energy_weight", "1.0",
    "--forces_weight", "100.0",
    "--swa_forces_weight", "10.0",
    "--loss", "huber",
    "--hidden_irreps", "128x0e + 128x1o + 128x2e",
    "--num_channels", "128",
    "--r_max", "5.0",
    "--batch_size", "16",
    "--valid_batch_size", "16",
    "--max_num_epochs", "500",
    "--swa",
    "--start_swa", "400",
    "--ema",
    "--ema_decay", "0.99",
    "--amsgrad",
    "--seed","1",
    "--scheduler_patience", "5",
    "--patience", "50",
    "--restart_latest",
    "--device", "cuda"
]

**Comment by ilyes319** [comment]
Hello,

i recommend you update your mace version to 0.3.13 and that you rename your keys in your xyz file to "REF_energy", "REF_forces" and "REF_stress", as the warning is saying it is unsafe to use the keys you are using. Then provide these keys to mace input with `--energy_key="REF_energy" --forces_key="REF_forces"`. If you still have a problem please then send me your log file with the newest mace version.

**Comment by imshivam24** [comment]
I have updated the MACE version and fixed the keys. Still, the problem persists. Attaching the log file for your perusal.
[MACE_model_regularized_run-1.log](https://github.com/user-attachments/files/20967933/MACE_model_regularized_run-1.log)

**Reply by ilyes319** [reply]
hello, can you tell me what is the source of your data, which code did you use? Could you also please share the plots that are saved as a training summary.
Please go over the troubleshooting doc and check you did not make any of these mistakes (including checking units, checking the sign of the gradient etc).

**Comment by imshivam24** [comment]
These are DFT optimization trajectories generated using VASP. Attached below are the plots. What could be the reasons for the exploding loss during training?
![MACE_model_regularized_run-1_train_Default_stage_one](https://github.com/user-attachments/assets/06fa2a99-9f0c-4e71-aef5-dc529651a42f)
![MACE_model_regularized_run-1_train_Default_stage_two](https://github.com/user-attachments/assets/a2ff1791-bf9f-4c8f-b012-247fea6f1689)

**Reply by ilyes319** [reply]
Clearly your energy and forces have not been loaded... Can you please send me a small part of your data so i can reproduce.

**Reply by imshivam24** [reply]
I am sharing with you the trimmed version of my training data.
[train_strip.zip](https://github.com/user-attachments/files/20977642/train_strip.zip)

**Reply by ilyes319** [reply]
i was not able to reproduce your problem, could you email at ib467@cam.ac.uk your train and validation set please. Thank you.

**Comment by imshivam24** [comment, marked as answer]
After a careful review of my dataset, I discovered a single datapoint with an anomalously high energy value. Upon removing this outlier from the dataset, the training proceeded as expected, and the loss values for training, validation, and test sets are now consistent.
